test/test_torch.py

```diff
@@ -2757,6 +2757,18 @@ def test_pickle_function(self):
             b = pickle.loads(serialized)
             self.assertEqual(a, b)
 
+        def test_norm_deprecated(self):
+            expected_message = (
+                'torch.norm and torch.functional.norm are deprecated. '
+                'Please use torch.linalg.norm instead.')
+            for func in [torch.norm, torch.functional.norm]:
+                if '__warningregistry__' in func.__globals__:
```

Comment:
When changing the warning to appear only once this test may or may not throw a warning, an unfortunate limitation of our testing framework. We have `self.maybeWarnsRegex` to handle this case. You can just call norm under that as a context manager.